hack/generator/pkg/astmodel/object_type.go

```diff
@@ -56,8 +56,19 @@ func (objectType *ObjectType) AsDeclarations(codeGenerationContext *CodeGenerati
 
 func (objectType *ObjectType) generateMethodDecls(codeGenerationContext *CodeGenerationContext, typeName TypeName) []ast.Decl {
 	var result []ast.Decl
-	for methodName, function := range objectType.functions {
-		funcDef := function.AsFunc(codeGenerationContext, typeName, methodName)
+
+	// Functions must be ordered by name for deterministic output
+	var functions []Function
+	for _, f := range objectType.functions {
+		functions = append(functions, f)
+	}
+
+	sort.Slice(functions, func(i int, j int) bool {
+		return functions[i].Name() < functions[j].Name()
+	})
+
+	for _, f := range objectType.functions {
```

Comment:
Shouldn't you be iterating over `functions` here, since that's the collection which was sorted?

Reply:
Although interestingly we weren't sorting these before and it was ok? They must have been being sorted elsewhere?

Reply:
Good catch, fixed to use the local `functions`.

The `AsFunc()` call creates an AST fragment, so there's no opportunity for sorting elsewhere.

This list contains directly implemented functions (any functions included to implement an interface are elsewhere), so I think we got lucky that there were only ever zero or one present.